content/agent/kubernetes/dogstatsd.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Sending Custom Metrics to DogStatsD for Kubernetes
+kind: documentation
+further_reading:
+- link: "agent/kubernetes/metrics"
+  tag: "documentation"
+  text: Kubernetes Metrics
+- link: "developers/dogstatsd"
+  tag: "documentation"
+  text: DogStatsD
+---
+
+To emit custom metrics from your Kubernetes application, use [DogStatsD][1], a metrics aggregation service bundled with the Datadog Agent. DogStatsD implements the [StatsD][2] protocol with some differences. Check out more on the [DogStatsD documentation][1].
+
+**Note**: You can use [DogStatsD over Unix Domain Socket][9].
+
+## Bind the DogStatsD port to a host port
+
+Add a `hostPort` to your `datadog-agent.yaml` file:
+
+```yaml
+ports:
+  - containerPort: 8125
+    hostPort: 8125
+    name: dogstatsdport
+    protocol: UDP
+```
+
+This enables your applications to send metrics via DogStatsD on port `8125` on whichever node they happen to be running.
+
+**Note**: `hostPort` functionality requires a networking provider that adheres to the [CNI specification][3], such as Calico, Canal, or Flannel. For more information, including a workaround for non-CNI network providers, consult the [Kubernetes documentation][4].
+
+To deploy the service, apply the change:
```

Comment:
services are a specific object type in k8s, this file deploys a DaemonSet object. Re-applying the spec after modification can be implied I guess